src/mumble/widgets/TrayIcon.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright 2024 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "TrayIcon.h"
+
+#include "../ClientUser.h"
+#include "../MainWindow.h"
+#include "../Global.h"
+
+#include <QApplication>
+
+TrayIcon::TrayIcon() : QSystemTrayIcon(Global::get().mw), m_statusIcon(Global::get().mw->qiIcon) {
+	applyIcon(m_statusIcon);
+
+	setToolTip("Mumble");
+
+	QObject::connect(Global::get().mw, &MainWindow::windowHighlighted, this, &TrayIcon::on_timer_triggered);
```

Comment:
```suggestion
	assert(Global::get().mw);
	QObject::connect(Global::get().mw, &MainWindow::windowHighlighted, this, &TrayIcon::on_timer_triggered);
```
(to make things go boom early, if someone creates the objects in the wrong order during startup)